micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMetersConventions.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2025 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.jvm;
+
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.binder.MeterConvention;
+import io.micrometer.core.instrument.binder.MeterConventions;
+
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryType;
+
+public interface JvmMetersConventions {
+
+    JvmMetersConventions DEFAULT = new JvmMetersConventions() {
+    };
+
+    default JvmMemoryMeterConventions jvmMemoryMeterConventions() {
+        return JvmMemoryMeterConventions.DEFAULT;
+    }
+
+    default JvmClassLoadingMeterConventions jvmClassLoadingMeterConventions() {
+        return JvmClassLoadingMeterConventions.DEFAULT;
+    }
+
+    interface JvmMemoryMeterConventions extends MeterConventions<MemoryPoolMXBean> {
+
+        JvmMemoryMeterConventions DEFAULT = new JvmMemoryMeterConventions() {
+        };
+
+        @Override
+        default Tags getCommonTags(MemoryPoolMXBean memoryPoolBean) {
+            return Tags.of("id", memoryPoolBean.getName(), "area",
+                    MemoryType.HEAP.equals(memoryPoolBean.getType()) ? "heap" : "nonheap");
+        }
+
+        default MeterConvention<MemoryPoolMXBean> getMemoryUsedConvention() {
+            return new MeterConvention<MemoryPoolMXBean>() {
```

Comment:
It doesn't especially matter for the particular meters where this will be used because they are registered once on binding. For other meters that are retrieved in order to record every time like GC timing metrics, it's wasteful to return a new object like this.